core/state/statedb_arbitrum.go

```diff
@@ -237,9 +234,13 @@ func (s *StateDB) StartRecording() {
 }
 
 func (s *StateDB) RecordProgram(moduleHash common.Hash) {
+	asm, err := s.TryGetActivatedAsm(moduleHash)
+	if err != nil {
+		log.Crit("can't find activated wasm while recording", "modulehash", moduleHash)
```

Comment:
Is the program guaranteed to be in cache here? I don't think we can trust that it's in the database, e.g. if the tx reverts.

Reply:
yes. It's called from arbos/programs/native.go:callProgram. Before we get to it we take the asm and module and store them at least to the statedb.